src/contexts/cartContext.js

```diff
@@ -0,0 +1,31 @@
+import { createContext, useContext, useReducer } from "react";
+import { cartReducer } from "../utils/CartReducer";
+import { useAxios } from "../utils/useAxios";
+import { useState, useEffect } from "react";
+
+const CartContext = createContext();
+
+const CartProvider = ({ children }) => {
+    const [cartState, cartDispatch] = useReducer(cartReducer, {
```

Comment:
here you can set state name userState like that bcoz you are not storing only cart details. here you store wishlist with cart.

Reply:
thanks for pointing out.